lib/IdPDisco.php

```diff
@@ -77,13 +81,22 @@ public function handleRequest()
             );
         }
 
+        // Get the SP's name
+        $spName = "";
```

Comment:
Do you want `$spName` to default to `""` or to `null`? It seems like `null` might be more accurate.